elementary/monitor/api/alerts/alerts.py

```diff
@@ -0,0 +1,277 @@
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=TestAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_model_alerts(self, days_back: int):
+        pending_model_alerts = self._query_pending_model_alerts(days_back)
+        last_alert_sent_times = self._query_last_model_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_model_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=ModelAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_source_freshness_alerts(self, days_back: int):
+        pending_source_freshness_alerts = self._query_pending_source_freshness_alerts(
+            days_back
+        )
+        last_alert_sent_times = self._query_last_source_freshness_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_source_freshness_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=SourceFreshnessAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def _sort_alerts(
+        self,
+        pending_alerts: AlertsQueryResult[Alert],
+        last_alert_sent_times: Dict[str, str],
+    ) -> Tuple[List[str], AlertsQueryResult[Alert]]:
+        suppressed_alerts = self._get_suppressed_alerts(
+            pending_alerts, last_alert_sent_times
+        )
+        alerts_to_skip = []
+        alerts_to_send = []
+        malformed_alerts_to_send = []
+
+        for alert in pending_alerts.alerts:
+            if alert.id in suppressed_alerts:
+                alerts_to_skip.append(alert.id)
+            else:
+                alerts_to_send.append(alert)
+
+        for alert in pending_alerts.malformed_alerts:
+            if alert.id in suppressed_alerts:
+                alerts_to_skip.append(alert.id)
+            else:
+                malformed_alerts_to_send.append(alert)
+
+        return alerts_to_skip, AlertsQueryResult(
+            alerts=alerts_to_send, malformed_alerts=malformed_alerts_to_send
+        )
+
+    def _get_suppressed_alerts(
+        self, alerts: AlertsQueryResult[Alert], last_alert_sent_times: Dict[str, str]
+    ) -> List[str]:
+        suppressed_alerts = []
+        current_time_utc = datetime.utcnow()
+        for alert in alerts.alerts:
+            id = (
+                alert.test_unique_id
+                if isinstance(alert, TestAlert)
+                else alert.unique_id
+            )
+            suppression_interval = alert.alert_suppression_interval
+            last_sent_time = (
+                datetime.fromisoformat(last_alert_sent_times.get(id))
+                if last_alert_sent_times.get(id)
+                else None
+            )
+            is_alert_in_suppression = (
+                (current_time_utc - last_sent_time).seconds / 3600
```

Comment:
if `last_sent_time` is `None`, this would throw an error.

Reply:
last_sent_time couldn't be None, because we add epoch time 0 when sent time does not exists (it is happening for typing issues)

Reply:
I can add a check just to make sure